samples/OpenIDConnectSample/Startup.cs

```diff
@@ -0,0 +1,56 @@
+﻿using Microsoft.AspNet.Builder;
+using Microsoft.AspNet.Http;
+using Microsoft.Framework.DependencyInjection;
+using Microsoft.AspNet.Security.OpenIdConnect;
+using Microsoft.AspNet.Http.Security;
+using Microsoft.AspNet.Security;
+
+namespace OpenIdConnectSample
+{
+    public class Startup
+    {
+        public void Configure(IApplicationBuilder app)
+        {
+            app.UseServices(services =>
+            {
+                services.AddDataProtection();
+                services.Configure<ExternalAuthenticationOptions>(options =>
+                {
+                    options.SignInAsAuthenticationType = OpenIdConnectAuthenticationDefaults.AuthenticationType;
```

Comment:
@brentschmaltz @Tratcher all these changes are great and we definitely need them (the OIDC configuration discovery is just amazing). But we also need a basic token validation middleware that behaves exactly like the existing OAuth2 bearer middleware and allows full pluggability without relying on complicated and OIDC-specific things. Here's my suggestions:

* Rename the existing OAuth2 bearer middleware to `Microsoft.AspNet.Security.Tokens.TokenAuthenticationMiddleware`: there's definitely nothing OAuth2/OIDC-specific with `OAuthBearerAuthenticationMiddleware` in its current form (almost identical to Katana 3) and this led to a terrible confusion. In Average Joe's mind, it should be clear that this middleware can be used with tokens issued by non-OAuth2 authorization servers: for instance, you must be able to plug in a `ISecureDataFormat` that relies on the data protection stuff or that communicates with a remote server to validate an opaque token that cannot be verified locally. That's also the place where I'd create the `ISecureDataFormat` -> `SecurityTokenHandler` adapter.

* Move your new OIDC-stuff to a new package and a new middleware subclassing `TokenAuthenticationMiddleware`... what about `Microsoft.AspNet.Security.Tokens.OpenIdConnect`?